WordPress/gutenberg
# [RNMobile] Synced Patterns: Ensure title is always visible

* Gutenberg Mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/6303

## What?

Fixes https://github.com/WordPress/gutenberg/issues/37203, by ensuring the title section of Synced Patterns is alway visible within the editor.

## Why?

There are currently certain scenarios where the title section of Synced Patterns is invisible. This happens if a block-based theme with a light background is active, as the native editor displays the light background in contrast with the block's dark mode styles. The same happens if a theme has a dark background and a device's light mode if enabled.

## How?

The logic associated with the title section's styles has been updated to use the block based theme's colour scheme.

## Testing Instructions

Follow the _preparation steps_ once for each mode (dark/light) and then proceed with the _testing steps_.

### Preparation steps

* Ensure the device is in either dark mode or light mode.
* Activate a theme with support for global styles, e.g. [Byrne](https://byrnedemo.wordpress.com/).
* In your site's /wp-admin area, navigate to the site editor (Appearance → Editor) and select a style corresponding to the opposite mode (dark background for light mode, and vice versa).
* Create a [synced pattern](https://wordpress.org/documentation/article/reusable-blocks/) with a title if you haven't already, while in the browser.
* In the app, navigate to the editor for any of the site's posts or pages.

### Testing steps

* Tap the + icon to open the block inserter and select the "Synced Patterns" tab.
* Add a synced pattern to the editor.
* Select the synced pattern and verify that you're able to see its icon, title, and the help button in the top bar.

### Testing Instructions for Keyboard

N/A, solely a visual change.

## Screenshots or screencast

### Dark mode with light background

| Before  | After |
| ------------- | ------------- |
| <img src="https://github.com/WordPress/gutenberg/assets/2998162/129c51a7-433a-4f07-8532-14c447e6fa4d" width="100%"> | <img src="https://github.com/WordPress/gutenberg/assets/2998162/96d1b293-9f51-44cf-b016-226a4e87f29e" width="100%"> |

### Light mode with dark background

| Before  | After |
| ------------- | ------------- |
| <img src="https://github.com/WordPress/gutenberg/assets/2998162/1c93c568-9965-4c49-a1be-01494e2c6843" width="100%"> | <img src="https://github.com/WordPress/gutenberg/assets/2998162/44cb18a7-2422-4f17-9bcb-bc948bb88a01" width="100%"> |
